### PR TITLE
Newsletter Signup Error Handling

### PR DESCRIPTION
**Pull request by robdivincenzo**
# Description

There currently is no error message returned to the user in the case that the 3rd party newsletter sign up service is down or does not return 201. This pull request will now display an API error to the user. These API Errors are different and separate from validation errors, and are treated as such.

Link to sample test page: https://foundation-s-11406-news-g7ddcn.mofostaging.net/en/test-the-invalid-newsletter-signup/
Related PRs/issues: #11406

Login: admin
Password: 4@IMcwJ!(GqoL9^Z

# To Test

An invalid signup snippet (non-existant newsletter) has been created and added to the sample test page. Navigate to that link and attempt to sign up. You should now receive an error message.

Repeat for any locale like /fr/ instead of /en/ and verify localization is occurring.

┆Issue is synchronized with this [Jira Story](https://mozilla-hub.atlassian.net/browse/TP-63)

### CODE REVIEW

File: source/js/components/newsletter-signup/organisms/with-submission-logic.jsx [hunk 1]
```diff
@@ -193,9 +193,19 @@ function withSubmissionLogic(WrappedComponent) {
         // [TODO] We will need to update this logic depending on what comes out of
```
Comment (mmmavis) [inline source/js/components/newsletter-signup/organisms/with-submission-logic.jsx:193]: This comment can finally be removed :D

**Review by mmmavis — CHANGES_REQUESTED** [review]
PR is working well. Thanks @robdivincenzo ! 

Two little things - 

- There are some comments in `with-submission-logic.jsx` that can be removed.
- Regarding the error message, do you think we can change it to something more generic? `Signup was not created.` and  `Unable to resolve signup.` feel quite technical and not so user friendly in my opinion. Maybe we can ask in the team channel to brainstorm copy ideas? 👀

**Review by mmmavis — CHANGES_REQUESTED** [review]
For localization, we are moving away from using `getText` as it's reading hardcoded strings in `source/js/components/petition/locale-strings.jsx` and not hooked up with our localization flow. I wonder if there's a way to expose the new strings (the two error messages we are adding in this PR) to Pontoon and have our backend to [send back the localized error message directly](https://github.com/MozillaFoundation/foundation.mozilla.org/blob/main/network-api/networkapi/campaign/views.py#L99-L102) ? @jhonatan-lopes @danielfmiranda

**Comment by jhonatan-lopes** [comment]
Hey @mmmavis! Localisation on JS is handled by serving `.mo` files to the client (see https://docs.djangoproject.com/en/4.2/topics/i18n/translation/#internationalization-in-javascript-code).

We have that enabled on our codebase already. See:

https://github.com/MozillaFoundation/foundation.mozilla.org/blob/bf8bd1d4116b5162e087742cd6d8cf85d7854db0/network-api/networkapi/urls.py#L120

and

https://github.com/MozillaFoundation/foundation.mozilla.org/blob/bf8bd1d4116b5162e087742cd6d8cf85d7854db0/network-api/networkapi/templates/pages/base.html#L135

It seems that from the JS perspective, you just need to call `gettext`, which will then issue a request to the server to fetch the relevant `.mo` file.

We are also prepared to generate messages from JS files:

https://github.com/MozillaFoundation/foundation.mozilla.org/blob/bf8bd1d4116b5162e087742cd6d8cf85d7854db0/tasks.py#L33-L45

I believe that covers all cases but if we have any further questions, I suggest getting in contact with Théo to confirm.

File: source/js/components/newsletter-signup/organisms/with-submission-logic.jsx (hunk 1 above)
Comment (robdivincenzo) [reply source/js/components/newsletter-signup/organisms/with-submission-logic.jsx]: You got it! 👍

**Comment by robdivincenzo** [comment]
@mmmavis @jhonatan-lopes Thanks for all the extra context related to localization!

Interestingly, I do not see strings exporting from JS, only from django (i.e through `blocktrans`). Though `makemessages` is running the correct `-d djangojs` parameter.

I'm tempted to include a hardcode `getText` string for this issue, but spin off a separate issue related just to looking at JS localization. I could use the already translated "Something went wrong" string from `source/js/components/petition/locale-strings.jsx` which would also address the error message being too technical. Thoughts?

**Comment by jhonatan-lopes** [comment]
Hey @robdivincenzo, are you running `inv makemessages`? If so, could this be broken due to #11825?

**Comment by robdivincenzo** [comment]
Hey @jhonatan-lopes! Yes, that's the command I ran to check for detecting JS strings. I was also using the updated command from that related issue's [PR](https://github.com/MozillaFoundation/foundation.mozilla.org/pull/12019). I could verify the .po files updated with a new Django string but not a new JS string. Is there an additional step in the workflow?

**Comment by jhonatan-lopes** [comment]
:thinking: that's really weird. We can check if it worked before on main but from a Django perspective it seems like everything is in order.

**Comment by robdivincenzo** [comment]
Based on slack discussions, we're going to approach it the old, hard-coded way to unblock this issue. We'll spin off a separate issue to look at djangojs and a second ticket to refactor this component's localization. We are also going with "Something went wrong and your signup wasn't completed. Please try again later." for the less technical error message per this slack convo: https://mozilla.slack.com/archives/C04DYP6T6VD/p1710357586006909

**Comment by robdivincenzo** [comment]
Hey @TheoChevalier, we have some hard-coded localized strings in this PR. Can you review the new translations located in `locale-strings.jsx`? Thank you!

**Comment by TheoChevalier** [comment]
Hi @robdivincenzo, I can get these translated by our translation agency alongside another project (hoping to be able to submit it later today or tomorrow)

Regarding your earlier question about using `window.gettext("My string.")` in jsx files, we just need to make this small change to the makemessages command in tasks.py:

Capture d’écran 2024-03-20 à 18 15 35

I’ve tried locally and was able to extract your string. The only caveat is that gettext doesn’t support backticks.

File: source/js/components/petition/locale-strings.jsx [hunk 2]
```diff
@@ -123,6 +125,8 @@ export default {
         und Ihre E-Mail-Einstellungen aktualisieren.
       </Fragment>
     ),
+    "Something went wrong and your signup wasn't completed. Please try again later.":
+      "Es ist ein Fehler aufgetreten und Ihre Anmeldung wurde nicht abgeschlossen. Bitte versuchen Sie es später noch einmal.",
```
Comment (TheoChevalier) [inline source/js/components/petition/locale-strings.jsx:129]:
```suggestion
      "Etwas ist schiefgelaufen und Ihre Anmeldung konnte nicht durchgeführt werden. Bitte versuchen Sie es erneut.",
```

File: source/js/components/petition/locale-strings.jsx [hunk 3]
```diff
@@ -184,6 +188,8 @@ export default {
         y actualizar tus preferencias de correo electrónico.
       </Fragment>
     ),
+    "Something went wrong and your signup wasn't completed. Please try again later.":
+      "Algo salió mal y tu registro no se completó. Por favor, inténtelo de nuevo más tarde.",
```
Comment (TheoChevalier) [inline source/js/components/petition/locale-strings.jsx:192]:
```suggestion
      "Algo salió mal y tu registro no se ha completado. Vuelve a intentarlo más tarde.",
```

File: source/js/components/petition/locale-strings.jsx [hunk 4]
```diff
@@ -242,6 +248,8 @@ export default {
         et modifier vos préférences de messagerie.
       </Fragment>
     ),
+    "Something went wrong and your signup wasn't completed. Please try again later.":
+      "Quelque chose s'est mal passé et votre inscription n'a pas été finalisée. Veuillez réessayer plus tard.",
```
Comment (TheoChevalier) [inline source/js/components/petition/locale-strings.jsx:252]:
```suggestion
      "Quelque chose s’est mal passé et votre inscription n’a pas été finalisée. Veuillez réessayer plus tard.",
```

File: source/js/components/petition/locale-strings.jsx [hunk 5]
```diff
@@ -299,6 +307,8 @@ export default {
         i zaktualizować preferencje.
       </Fragment>
     ),
+    "Something went wrong and your signup wasn't completed. Please try again later.":
+      "Coś poszło nie tak i Twoja rejestracja nie została ukończona. Spróbuj ponownie później.",
```
Comment (TheoChevalier) [inline source/js/components/petition/locale-strings.jsx:311]:
```suggestion
      "Coś poszło nie tak i rejestracja nie została ukończona. Spróbuj ponownie później.",
```

File: source/js/components/petition/locale-strings.jsx [hunk 6]
```diff
@@ -357,5 +367,7 @@ export default {
         e atualizar suas preferências de e-mail.
       </Fragment>
     ),
+    "Something went wrong and your signup wasn't completed. Please try again later.":
+      "Algo deu errado e sua inscrição não foi concluída. Por favor, tente novamente mais tarde.",
```
Comment (TheoChevalier) [inline source/js/components/petition/locale-strings.jsx:371]:
```suggestion
      "Algo deu errado e sua assinatura não foi concluída. Tente novamente mais tarde.",
```

**Review by TheoChevalier — CHANGES_REQUESTED** [review]
Hey @robdivincenzo added the updated translations

**Review by TheoChevalier — APPROVED** [review]
Thanks, looks good to me!

**Review by mmmavis — APPROVED** [review]
Yay! Thanks for the awesome work, @robdivincenzo !

**Pull request merged**